src/client/InteractionClient.js

```diff
@@ -0,0 +1,235 @@
+'use strict';
+
+const BaseClient = require('./BaseClient');
+const ApplicationCommand = require('../structures/ApplicationCommand');
+const CommandInteraction = require('../structures/CommandInteraction');
+const { Events, ApplicationCommandOptionType, InteractionType, InteractionResponseType } = require('../util/Constants');
+
+let sodium;
+
+function transformCommand(command) {
+  return {
+    name: command.name,
+    description: command.description,
+    options: command.options?.map(function m(o) {
+      return {
+        type: ApplicationCommandOptionType[o.type],
```

Comment:
```suggestion
        type: o.type,
```

DiscordAPI takes integer for the option type (source: https://discord.com/developers/docs/interactions/slash-commands#applicationcommandoption)

Currently a DiscordAPIError is thrown